src/passes/ReorderGlobals.cpp

```diff
@@ -133,17 +133,26 @@ struct ReorderGlobals : public Pass {
     }
 
     // Compute dependencies.
-    Dependencies deps;
+    std::vector<std::unordered_set<size_t>> dependsOn(globals.size()),
+      dependedUpon(globals.size());
     for (Index i = 0; i < globals.size(); i++) {
       auto& global = globals[i];
       if (!global->imported()) {
         for (auto* get : FindAll<GlobalGet>(global->init).list) {
           auto getIndex = originalIndices[get->name];
-          deps.dependsOn[i].insert(getIndex);
-          deps.dependedUpon[getIndex].insert(i);
+          dependsOn[i].insert(getIndex);
+          dependedUpon[getIndex].insert(i);
         }
       }
     }
+    Dependencies deps;
+    deps.dependsOn.reserve(globals.size());
+    deps.dependedUpon.reserve(globals.size());
+    for (size_t i = 0; i < globals.size(); ++i) {
+      deps.dependsOn.emplace_back(dependsOn[i].begin(), dependsOn[i].end());
+      deps.dependedUpon.emplace_back(dependedUpon[i].begin(),
+                                     dependedUpon[i].end());
+    }
```

Comment:
Can we fill in `deps` directly on lines 143-144? I am not sure I see why we need the temporary structures that are declared on lines 136-137.

Reply:
Oh, is this converting from a set to a vector? That's what I missed before I think.